offload/DeviceRTL/src/Utils.cpp

```diff
@@ -57,6 +59,12 @@ int32_t shuffleDown(uint64_t Mask, int32_t Var, uint32_t LaneDelta,
   return __builtin_amdgcn_ds_bpermute(Index << 2, Var);
 }
 
+uint32_t ballotSync(uint32_t Mask, int32_t Pred) {
```

Comment:
The wavefront size for AMDGPU can be 32 or 64, which means the mask here should be `uint32_t` or `uint64_t` correspondingly. Do we want to unify them with `uint64_t` and provide a full mark variable, or provide `_32` and `_64` variant? @jdoerfert

Reply:
64 bit seems reasonable, I think.

Reply:
Just do what https://github.com/llvm/llvm-project/blob/main/libc/src/__support/GPU/amdgpu/utils.h#L131 does.

I want to put a lot of these utilities in compiler-rt or something, since we have multiple users.